### PR TITLE
Add routine for sparse vector addition

### PR DESCRIPTION
**Pull request by FreyJo**
```
// z += alpha * x[idx]
void blasfeo_dvec_add_sp_in(int m, double alpha, int *idx, struct blasfeo_dvec *sx, int xi, struct blasfeo_dvec *sz, int zi);
```
Maybe should be renamed.

### CODE REVIEW

**Comment by giaf** [comment]
Thanks for the proposal of adding this additional auxiliary routine.
I would suggest to call the routine `blasfeo_dvecexad_sp` (i.e. vector extract and add - sparse variant).
I think this would be more consistent with the existing routines, and when I have some time I would possibly also rename `blasfeo_dvecad_sp` into `blasfeo_dvecinad_sp`, but that's another story.

The new routine should be added to some additional files.
The following files seem to be missing:
- auxiliary/s_aux_lib4.c
- auxiliary/s_aux_lib8.c
- include/blasfeo_s_aux.h
- include/blasfeo_s_aux_ref.h

If you can make these changes, then I can merge the PR straight away.

**Comment by FreyJo** [comment]
Thanks for checking!
It is all addressed in the last commit.

File: auxiliary/d_aux_ref.c [hunk 1]
```diff
@@ -173,6 +174,7 @@
 #define VECAD_SP blasfeo_dvecad_sp
 #define VECIN_SP blasfeo_dvecin_sp
 #define VECEX_SP blasfeo_dvecex_sp
+#define VECEXAD_SP blasfeo_dvecexad_sp
```
Comment (giaf) [inline auxiliary/d_aux_ref.c:177]: this should also go in the single-precision counterpart `s_aux_ref.c`

**Comment by giaf** [comment]
Two more small fixes and then we should be good.
With all the different targets and backends each single routine ends up in a lot of different places :p

**Comment by FreyJo** [comment]
yes, now it should be fine.
>  With all the different targets and backends each single routine ends up in a lot of different places :p

Great that you still have the overview :D

**Comment by giaf** [comment]
I try my best :p
Here it was mainly a matter of symmetry with the double precision version that was already correct.

**Pull request merged**